ast2vec/tests/test_cloning.py

```diff
@@ -0,0 +1,128 @@
+            args = copy.copy(self.base_args)
+            args.output = tmpdir
+            args.linguist = tests.ENRY
+            args.languages = ["Python"]
+            self.assertIsNone(clone_repositories(args))
+            self._validate_clone(tmpdir, args)
+
+    def test_multiple_repositories(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            args = copy.copy(self.base_args)
+            args.output = tmpdir
+            args.linguist = tests.ENRY
+            args.languages = ["Python"]
+            args.input.append(os.path.join(os.path.dirname(__file__), "test_repos_list.txt"))
+            self.assertIsNone(clone_repositories(args))
+            self._validate_clone(tmpdir, args)
+
+    def test_repo_creation(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            args = copy.copy(self.base_args)
+            args.output = str.encode(tmpdir)
+            args.linguist = None
+            with self.assertRaises(TypeError):
+                clone_repositories(args)
+
+    def test_prepare_repo_dir(self):
+        with tempfile.TemporaryDirectory() as tmpdir:
+            for repo_url in self.repo_urls:
+                repo_url = RepoCloner._prepare_repo_url(repo_url)
+                self.assertEqual(
+                    os.path.join(tmpdir, "cool@you"),
+                    RepoCloner._prepare_repo_dir(repo_url, tmpdir, True))
+
+    def test_prepare_repo_url(self):
+        results = [
+            "https://whatever.cite.com/cool/you.git",
+            "https://whatever.cite.com/cool/you.git",
+            "https://whatever.cite.com/cool/you",
+            "https://whatever.cite.com/cool/you",
+            "https://whatever.cite.com/cool/you",
+            "https://whatever.cite.com/cool/you.git",
+        ]
+        for correct_url, repo_url in zip(results, self.repo_urls):
+            self.assertEqual(correct_url, RepoCloner._prepare_repo_url(repo_url))
+
+    def _collect_repo_files(self, repo_dir):
+        files = list()
+        for dirname, dirnames, filenames in os.walk(str.encode(repo_dir)):
+            for filename in filenames:
+                full_filename = os.path.join(dirname, filename)
+                files.append(full_filename)
+        return files
+
+    def _validate_clone(self, tmpdir, args):
+        self.assertTrue(len(os.listdir(args.output)) > 0)
+
+        if args.linguist:
+            enry = args.linguist
+            self.assertTrue(os.path.isfile(enry))
+            self.assertEqual(os.stat(enry).st_mode & 0o777, 0o777)
+            output = subprocess.check_output([enry, "-json", args.output])
+            lang_files = json.loads(output.decode("utf-8"))
+
+            files = [str.encode(os.path.join(args.output, f)) for files in
+                     lang_files.values() for f in files]
+            repo_files = self._collect_repo_files(args.output)
+            self.assertEqual(set(files), set(repo_files))
+
+            if hasattr(args, 'languages'):
```

Comment:
quotes
@vmarkovtsev , we definitely need to share some kind of script that will check style. It's so easy to miss quotes